### DISCUSSION TITLE
[Question] Should we add slice support to commonly used data structures

### DISCUSSION THREAD

**Discussion by dylech30th** [dotnet/runtime#67703] [Q&A] [answered]
The new slice pattern requires the object to be sliceable, but most data structures that are preferred by people over arrays such as `List` do not compliant with this requirement, and since the type class, a.k.a., [roles](https://github.com/dotnet/csharplang/issues/5497) still has a long way to go, so maybe this gap should be compensated by adding corresponding methods, or by allowing the extension `Slice` method?

**Comment by 333fred** [comment, marked as answer]
This is a question for the runtime, so I'm moving it over there. See also https://github.com/dotnet/runtime/issues/66773.
